pkg/server/config/config.go

```diff
@@ -16,6 +18,17 @@ func Load(file string) (conf Config, err error) {
 	}
 
 	if err = yaml.Unmarshal(data, &conf); err != nil {
+		// Ensure we can and should inspect the AuthenticatingProxy member
+		if conf.AuthenticatingProxy != nil && len(conf.AuthenticatingProxy.AllowedSpiffeIds) > 0 {
```

Comment:
This might be a go specific question. If unmarshal fails, would config always take on the `Config` struct? For example, if we have a field which name does not match any of the ones specified in `Config`, I don't think we would be able to do additional check on it in that case

Reply:
Would it be helpful to also check the validity of the spiffe IDs even if unmarshalling is successful? Or would the error from `Unmarshal` always encapsulate all possible spiffe erros?

Reply:
To your first question, because `conf` is defined as type `Config` it is by default empty and the common pattern when umarshalling is to take the pointer of the struct that will be used to unmarshal the data (see [ref](https://go.dev/ref/spec#The_zero_value)).

I am curious about your second question, though. I also don't know if we shouldn't run validation on successful unmarshalling.